shadows/framework/src/main/java/org/robolectric/shadows/ShadowActivityManager.java

```diff
@@ -317,8 +319,11 @@ public void addApplicationExitInfo(String processName, int pid, int reason, int
   }
 
   /** Adds given {@link ApplicationExitInfo}, see {@link ApplicationExitInfoBuilder} */
-  public void addApplicationExitInfo(ApplicationExitInfo info) {
-    appExitInfoList.addFirst(info);
+  @RequiresApi(api = R)
+  public void addApplicationExitInfo(Object info) {
+    if (info instanceof ApplicationExitInfo) {
+      appExitInfoList.addFirst(info);
```

Comment:
nit: just cast in order to throw a ClassCastException if the wrong type is provided?

```appExitInfoList.addFirst((ApplicationExitInfo) info);```

Reply:
There is type checking at`if` statement, and the IDE shows the casting is redundant. What about throws `RuntimeException` at `else` statement?

Reply:
Done with `Preconditions.checkArgument`.